packages/cardano-services/src/TxWorker/loadTxWorker.ts

```diff
@@ -12,12 +11,17 @@ export interface TxWorkerArgs {
 }
 
 export const loadTxWorker = async (args: TxWorkerArgs) => {
-  const { loggerMinSeverity, ogmiosUrl, rabbitmqUrl } = args.options;
-  const txSubmitProvider = ogmiosTxSubmitProvider(urlToConnectionConfig(ogmiosUrl));
+  const { loggerMinSeverity, serviceDiscoveryBackoffFactor, serviceDiscoveryTimeout } = args.options;
   const logger = createLogger({ level: loggerMinSeverity, name: 'tx-worker' });
+  const dnsResolver = createDnsResolver(
+    {
+      factor: serviceDiscoveryBackoffFactor,
+      maxRetryTime: serviceDiscoveryTimeout
+    },
+    logger
+  );
+  const txSubmitProvider = await getOgmiosTxSubmitProvider(dnsResolver, args.options);
+  const url = await getRabbitMqUrl(dnsResolver, args.options);
 
-  // Ensure rabbitmqUrl is not undefined
-  if (!rabbitmqUrl) throw new MissingProgramOption(ServiceNames.TxSubmit, CommonOptionDescriptions.RabbitMQUrl);
-
-  return new TxSubmitWorker({ ...args.options, rabbitmqUrl }, { logger, txSubmitProvider });
+  return new TxSubmitWorker({ ...args.options, rabbitmqUrl: url }, { logger, txSubmitProvider });
```

Comment:
We might want to also wrap this in a proxy - re-resolve if rabbitmq connection of the worker fails

Reply:
Yep makes sense, just need to verify which method needs to be intercepted by the proxy, I guess `start()` where connects to the rabbitmq. 
Do you think we should introduce it within this PR or as a smaller separate task? @mkazlauskas

Reply:
It can fail after starting though. Maybe `this.#channel?.get` :thinking: Let's do this as separate task, maybe @iccicci can help to figure out the best way to do this for rabbitmq